fix(linter): `eslint/radix` rule correctly check for unbound symbols

overlookmotel opened the pull request:
`SymbolTable::get_symbol_id_from_name(name).is_none()` is not always correct, because it will return `false` if there is a binding *anywhere* in the AST with that name, whereas what we actually want to know is whether *this* `IdentifierReference` is referring to a global or not.

Instead, look up whether this reference is resolved or not using `SymbolTable::is_global_reference`.

The 3 test cases added were not working prior to this change.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5446?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5447** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5447?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#5446** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5446?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5445** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5445?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_linter/src/rules/eslint/radix.rs:
```diff
@@ -70,7 +70,7 @@ impl Rule for Radix {
                 Expression::Identifier(ident) => {
                     if ident.name == "parseInt"
-                        && ctx.symbols().get_symbol_id_from_name("parseInt").is_none()
+                        && ctx.symbols().is_global_reference(ident.reference_id().unwrap())
```
Is there ever a case where `reference_id()` would return `None`?

overlookmotel replied:
They are all `None` before semantic runs, and then all are always `Some` after semantic. Linter runs after semantic, so this `unwrap()` can never fail.

I would like to replace all these `Option<Id>`s with just plain `Id`s with 0 as a "sentinel value" equivalent to `None`, so we can avoid all this unwrapping (`ScopeId`, `SymbolId`, `ReferenceId`). But that's another issue...

overlookmotel committed:
fix(linter): `eslint/radix` rule correctly check for unbound symbols (#5446)

`SymbolTable::get_symbol_id_from_name(name).is_none()` is not always correct, because it will return `false` if there is a binding *anywhere* in the AST with that name, whereas what we actually want to know is whether *this* `IdentifierReference` is referring to a global or not.

Instead, look up whether this reference is resolved or not using `SymbolTable::is_global_reference`.

The 3 test cases added were not working prior to this change.
